OMCompiler/Compiler/NBackEnd/Util/NBDifferentiate.mo

```diff
@@ -1037,9 +1038,10 @@ public
             // in the case of recursive differentiation (e.g. function calls itself)
             dummy_func    := func;
             node.cls      := Pointer.create(new_cls);
-            node.name     := NBVariable.FUNCTION_DERIVATIVE_STR + "." + node.name;
+            der_func_name := NBVariable.FUNCTION_DERIVATIVE_STR + intString(listLength(func.derivatives));
```

Comment:
very nice idea! should be unique and work without additional index handling stuff.